tests/cpp_unit_tests/test_serializer.cpp

```diff
@@ -49,11 +49,11 @@ constexpr std::string_view single_dataset_list_indent =
   "type": "update",
   "is_batch": false,
   "attributes": {
-    "asym_load": [
+    "sym_load": [
       "id",
       "p_specified"
     ],
-    "sym_load": [
+    "asym_load": [
```

Comment:
i guess this is because the new implementation uses a `constexpr std::array` instead of a `std::map` with sorted strings as keys. i don't mind keeping it like this but i perfer the sorted behaviour (since most json formatters also sort their entries).

if we want to keep the sorted behaviour, then we need to change it in the serializer because changing it in the code generator would cause a breaking change for every time we add a new attribute

Reply:
Actually, I prefer to sort based on given order by `AllComponents`. In PGM not all components are equally important. We define the order in `AllComponents` as their importance to the physical model.

https://github.com/PowerGridModel/power-grid-model/blob/daa46892da9793536d7537f137da66ff51cfecd6/power_grid_model_c/power_grid_model/include/power_grid_model/all_components.hpp#L28-L30